### PR TITLE
Bug 1812843 - Increased the width for a CustomTabsToolbarFeature action button

### PR DESCRIPTION
**Pull request by t-p-white**
This bug was initially raised for the "Tweet" button.

Original image of bug provided:

"Tweet" has since be rebranded as "Post" but the problem is still the same.

Example of current Fenix:

Example in Chrome:

Example in Edge:

---

This patch aims to follow suit with the Chrome & Edge style by:

1.  increasing the width (doubling) to create a rectangle instead of a square - this will cater better for text
2. adding a small margin to the start of the button to prevent the URL from being too close, see example below

With margin:

Without margin:

---

With action button:

No action button:

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
3. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
4. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
5. Click on `View task in Taskcluster` in the new `DETAILS` section.
6. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1812843

### CODE REVIEW

**Comment by t-p-white** [comment]
I don't love that the margin has to get passed as much as it needs to with this approach, open to alternative suggestions

File: android-components/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt [hunk 1]
```diff
@@ -254,16 +254,16 @@ class BrowserToolbar @JvmOverloads constructor(
...
      *
      * If there is not enough room to show all icons then some icons may be moved to an overflow
      * menu.
      *
      * Related:
      * https://developer.mozilla.org/en-US/Add-ons/WebExtensions/user_interface/Browser_action
      */
-    override fun addBrowserAction(action: Toolbar.Action) {
-        display.addBrowserAction(action)
+    override fun addBrowserAction(action: Toolbar.Action, marginStart: Int?) {
```
Comment (MozillaNoah) [inline android-components/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt:265]: does this signature also need `marginStart: Int? = null`?

**Review by MozillaNoah — COMMENTED** [review]
So two thoughts with alternative approaches:
- Expand the margin API so it's all 4 directions and not just `start`.
- Make it an optional API in `Toolbar.ActionButton` or `Toolbar.Action`, and then function signatures would not need to be changed.

File: android-components/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt (hunk 1 above)
Comment (t-p-white) [reply android-components/components/browser/toolbar/src/main/java/mozilla/components/browser/toolbar/BrowserToolbar.kt:265]: As this is an overriding function:
1.  will inherit the default value from the interface
2. it cannot specify its own default

**Comment by t-p-white** [comment]
Thanks for the conversation @MozillaNoah 🙂  I've updated the PR with a different solution which I think is a cleaner approach than my original

File: android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml [hunk 2]
```diff
@@ -29,10 +29,12 @@
         android:layout_width="0dp"
```
Comment (t-p-white) [inline android-components/components/browser/toolbar/src/main/res/layout/mozac_browser_toolbar_displaytoolbar.xml:29]: we need to change the margin depending on whether the action button is displayed or not, see details for demo

**Review by MozillaNoah — APPROVED** [review]
Nice use of `layout_goneMarginEnd`. I totally forgot that was a thing with ConstraintLayout.

**Comment by t-p-white** [comment]
@MozillaNoah since your last review, I had to also update the `CustomTabsToolbarFeatureTest` to account for the update to the width value

**Pull request merged**